float64 leader_x
float64 leader_y
float64 leader_vel